### PR TITLE
docs: polish up output config

### PR DESCRIPTION
**Pull request by fi3ework**
## Summary

Polish up `output` documentation. The main target is to align with webpack's document if the abilities are aligned. Put legacy / tend to be deprecated fields to the end of the page and only provide a link to its substitute. 

### Configuration not aligned with webpack
- do not support `output.charset`
- do not support `output.chunkLoadTimeout`
- `output.clean` only supports boolean
- do not support `output.compareBeforeEmit`
- do not support `output.cssHeadDataCompression`
- do not support `output.ignoreBrowserWarnings`
- do not support `output.importMetaName`
- do not support `output.sourcePrefix`
- do not support `output.trustedTypes.onPolicyCreationFailure`
- `output.publicPath` only supports string
- do not support `strictModuleExceptionHandling` (deprecated)
- `output.assetModuleFilename` only supports string

I wonder, do we need to create a centralized issue to collect all config that misaligned with webpack and track in the future.

## Checklist

- [ ] Tests updated (or not required).
- [x] Documentation updated (or not required).

### CODE REVIEW

**Comment by chenjiahan** [comment]
> do we need to create a centralized issue to collect all config that misaligned with webpack and track in the future.

Agreed, it would be helpful if we could maintain such an issue to track the alignment progress.

File: website/docs/en/config/output.mdx [hunk 1]
```diff
@@ -877,9 +925,9 @@ The **return value of your entry point** will be assigned to `this` under the pr
 ```javascript
 this['MyLibrary'] = _entry_return_;
 
-// In a separate script...
+// 在一个单独的脚本中
 this.MyLibrary.doSomething();
-MyLibrary.doSomething(); // if `this` is window
+MyLibrary.doSomething(); // 如果 `this` 为 window 对象
```
Comment (chenjiahan) [inline website/docs/en/config/output.mdx:930]:
```suggestion
MyLibrary.doSomething(); // if `this` is window
```

File: website/docs/en/config/output.mdx [hunk 2]
```diff
@@ -1024,78 +1097,146 @@ Consumption (ESM):
...
+##### type: 'amd'
 
-This exposes your library under all the module definitions, allowing it to work with CommonJS, AMD, and as global variable. Take a look at the [UMD Repository](https://github.com/umdjs/umd) to learn more.
+This will expose your library as an AMD module.
 
-In this case, you need the `library.name` property to name your module:
+AMD modules require that the entry chunk (e.g. the first script loaded by the `<script>` tag) be defined with specific properties, such as to `define` and `require` which is typically provided by RequireJS or any compatible loaders (such as almond). Otherwise, loading the resulting AMD bundle directly will result in an error like `define is not defined`.
 
-```javascript
+With the following configuration...
```
Comment (chenjiahan) [inline website/docs/en/config/output.mdx:1110]:
```suggestion
With the following configuration:
```

File: website/docs/en/config/output.mdx [hunk 3]
```diff
@@ -1148,33 +1289,84 @@ System.register(
...
+      export: 'default',
+    },
   },
+};
+```
+
+The default export of your entry point will be assigned to the library name:
+
+```js
+// 如果你的库有默认导出
```
Comment (chenjiahan) [inline website/docs/en/config/output.mdx:1340]:
```suggestion
// If your library has a default export
```

File: website/docs/en/config/output.mdx [hunk 4]
```diff
@@ -1433,60 +1699,80 @@ module.exports = {
...
-- **Type:** `string`
-- **Default:** `"[runtime].[fullhash].hot-update.json"`
+````js title="rspack.config.js"
+module.exports = {
+  //...
+  output: {
+    workerWasmLoading: 'fetch',
+  },
+};
+```import { de } from 'date-fns/locale';
```
Comment (chenjiahan) [inline website/docs/en/config/output.mdx:1752]: unused code

File: website/docs/zh/config/output.mdx [hunk 5]
```diff
@@ -5,168 +5,199 @@ import { ApiMeta, Stability } from '../../../components/ApiMeta';
...
-  output: {
-    filename: '[contenthash].bundle.js',
-  },
-};
-```
+- **类型：** `false | 'array-push' | 'commonjs' | 'module' | string`
+- **默认值：** 根据 [`target`](/config/target) 和 [`output.module`](#outputmodule) 的配置推断
 
-当然你也可以组合使用：
+chunk 产物的格式，一般来说，target 为 web 或 Webworker 时会使用 `'array-push'` 格式；为 ESM 时会使用 `'module'` 格式；为 node.js 时会使用 `'commonjs'` 格式，插件可能会添加其他格式。
```
Comment (chenjiahan) [inline website/docs/zh/config/output.mdx:78]:
```suggestion
chunk 产物的格式，一般来说，target 为 web 或 webworker'时会使用 `'array-push'` 格式；为 ESM 时会使用 `'module'` 格式；为 node.js 时会使用 `'commonjs'` 格式，插件可能会添加其他格式。
```

File: website/docs/en/config/output.mdx [hunk 6]
```diff
@@ -26,147 +27,181 @@ The name of the file to be output by the Asset module. This value can be overrid
...
-module.exports = {
-  output: {
-    filename: '[contenthash].bundle.js',
-  },
-};
-```
+- **Type:** `false | 'array-push' | 'commonjs' | 'module' | string`
+- **Default:** Determined by [`target`](/config/target) and [`output.module`](#outputmodule)
+
+The format of chunks (formats included by default are `'array-push'` (web/WebWorker), `'commonjs'` (node.js), `'module'` (ESM), but others might be added by plugins).
```
Comment (chenjiahan) [inline website/docs/en/config/output.mdx:78]:
```suggestion
The format of chunks (formats included by default are `'array-push'` (web/webworker), `'commonjs'` (node.js), `'module'` (ESM), but others might be added by plugins).
```

File: website/docs/zh/config/output.mdx [hunk 7]
```diff
@@ -220,239 +252,195 @@ module.exports = {
...
 
-当 `target` 为 `'web'` 时，Rspack 会动态创建 `<script>` 和 `<link>` 标签来加载异步的 JavaScript 和 CSS 资源。如果这些资源的 URL 在其他域名下，且 `crossOriginLoading` 不为 `false`，那么 Rspack 会为 `<script>` 和 `<link>` 标签添加 `crossorigin` 属性。
+## output.enabledWasmLoadingTypes
 
-### 可选值
+- **类型：** `('fetch-streaming' | 'fetch' | 'async-node' | string | false)[]`
+- **默认值：** 根据 [`output.wasmLoading`](#outputwasmloading) 及 [`output.workerWasmLoading`](#workerWasmLoading) 的配置推断
 
-`crossOriginLoading` 有以下可选值：
+开启可用的 wasm 加载类型的运行时模块打包。
```
Comment (chenjiahan) [inline website/docs/zh/config/output.mdx:299]:
```suggestion
开启可用的 Wasm 加载类型的运行时模块打包。
```

File: website/docs/zh/config/output.mdx [hunk 8]
```diff
@@ -220,239 +252,195 @@ module.exports = {
...
+    filename: '[name].bundle.js',
   },
 };
 ```
 
-## output.publicPath
-
-- **类型：** `string`
-- **默认值：** targets 为 `'web'` 或 `'webworker'` 时为 `'auto'`
+使用内部 chunk id
```
Comment (chenjiahan) [inline website/docs/zh/config/output.mdx:391]:
```suggestion
使用内部 chunk id：
```

File: website/docs/zh/config/output.mdx [hunk 9]
```diff
@@ -487,84 +475,62 @@ module.exports = {
...
-## output.enabledLibraryTypes
+当你想在实际文件名中使用占位符时，可以过滤掉占位符替换。例如，要输出一个文件 `[name].js`，你必须通过在括号之间添加反斜杠来转义 `[name]` 占位符。这样 `[\name\]` 会生成 `[name]` 而不是被替换为资源的名称。
 
-- **类型：** `string[]`
+示例：`[\id\]` 会生成 `[id]` 而不是被替换为 id。
 
-入口点可用的 library 类型列表。
+如果使用函数来设置这个选项，函数将会接收一个包含上表中替换数据的对象。替换也会应用到返回的字符串中。传递的对象将具有这种类型：（属性取决于上下文）
 
 ```javascript
```
Comment (chenjiahan) [inline website/docs/zh/config/output.mdx:508]:
````
```ts
````

File: website/docs/zh/config/output.mdx [hunk 10]
```diff
@@ -605,45 +571,116 @@ module.exports = {
...
 
-可选的 salt，用于更新哈希值。
+<ApiMeta addedVersion={'0.3.5'} />
+
+- **类型：** `string`
+- **默认值：** `"webpackHotUpdate" + output.uniqueName`
+
+仅当 [`target`](/config/target) 设置为 `"web"` 时使用，它使用 JSONP 来加载热更新。
+
+JSONP函数用于异步加载热更新块。
```
Comment (chenjiahan) [inline website/docs/zh/config/output.mdx:636]:
```suggestion
JSONP 函数用于异步加载热更新块。
```

**Review by chenjiahan — APPROVED** [review]
🎉

**Comment by fi3ework** [comment]
@chenjiahan Thanks for your careful review. I'll trying to do better next time. 😆

**Pull request merged**